Empire/Empire.cpp

```diff
@@ -2800,6 +2800,42 @@ void Empire::CheckProductionProgress() {
         elem.progress += elem.allocated_pp;   // add allocated PP to queue item
         elem.progress_memory = elem.progress;
         elem.blocksize_memory = elem.blocksize;
+        
+        std::string build_description;
+        switch (elem.item.build_type) {
+            case BT_BUILDING: {
+                build_description = "BuildingType " + elem.item.name;
+                break;
+            }
+            case BT_SHIP: {
+                build_description = "Ships(s) with design id " + boost::lexical_cast<std::string>(elem.item.design_id);
+                break;
+            }
+            default: 
+                build_description = "unknown build type";
+        }
+
+        TemporaryPtr<UniverseObject> build_location = GetUniverseObject(elem.location);
+        if (!build_location || (elem.item.build_type == BT_BUILDING && build_location->ObjectType() != OBJ_PLANET)) {
+            ErrorLogger() << "Couldn't get valid build location for completed " << build_description;
+            continue;
+        }
+        TemporaryPtr<System> system = GetSystem(build_location->SystemID());
+        // TODO: account for shipyards and/or other ship production
+        // sites that are in interstellar space, if needed
+        if (!system) {
+            ErrorLogger() << "Empire::CheckProductionProgress couldn't get system for producing new " << build_description;
+            continue;
+        }
+
+        // check location condition before each item is created, so
+        // that items being produced can prevent subsequent
+        // completions on the same turn from going through
+        if (!this->ProducibleItem(elem.item, elem.location)) {
+            DebugLogger() << "Location test failed for " << build_description << " at location " << build_location->Name();
+            break;
```

Comment:
The code you moved to here was in a switch block, but this code is not, so I think this should be continue, not break, as break here will exit the for loop over queue contents.